human-forecasts/performance-board/app.R

```diff
@@ -0,0 +1,180 @@
+                  type = "deaths")
+
+cases_inc <- data.table::fread(here::here("data", "weekly-incident-cases.csv")) %>%
+    dplyr::mutate(inc = "incident", 
+                  type = "cases")
+
+
+obs <- dplyr::bind_rows(deaths_inc, cases_inc) %>%
+    dplyr::rename(true_value = value) %>%
+    dplyr::mutate(target_end_date = as.Date(target_end_date)) %>%
+    dplyr::filter(epiweek >= lubridate::epiweek(Sys.Date()) - 10)
+
+combined <- dplyr::inner_join(obs, forecasts)
+
+combined_full <- dplyr::left_join(obs, forecasts)
+
+
+
+scores <- scoringutils::eval_forecasts(combined, 
+                                       summarise_by = "forecaster_id") %>%
+    dplyr::mutate(forecaster_id = as.character(forecaster_id))
+
+
+scores_coverage <- scoringutils::eval_forecasts(combined, 
+                                                summarise_by = c("forecaster_id", 
+                                                                 "range", 
+                                                                 "quantile")) %>%
+    dplyr::mutate(forecaster_id = as.factor(forecaster_id))
+
+forecasters <- c("all", 
+                 unique(forecasts$forecaster_id))
+
+
+# Define UI for application that draws a histogram
+ui <- fluidPage(
+
+    # Application title
+    titlePanel("Performance Board"),
+    
+    fluidRow(column(2, 
+                    selectInput("forecaster_selection", 
+                                "Select a forecaster", 
+                                choices = forecasters)),
+             column(10, 
+                    fluidRow(column(12, 
+                                    h2("All applaud the glorious leader"),
+                                    tableOutput("table"))),
+                    
+                    h3("Score overview"), 
+                    plotOutput("scoretable"), 
+                    
+                    fluidRow(h3("Predictions vs. observed values")),
+                    fluidRow(column(6, 
+                                    h3("Germany"), 
+                                    plotOutput("predictions_germany")), 
+                             column(6,
+                                    h3("Poland"), 
+                                    plotOutput("predictions_poland"))),
+                    
+                    fluidRow(column(6, 
+                                    h3("Interval coverage"), 
+                                    plotOutput("interval_coverage")), 
+                             column(6, 
+                                    h3("Quantile coverage"), 
+                                    plotOutput("quantile_coverage")))
+                    
+                    )
+             )
+    
+    
+    
+         
+)
+
+
+
+# Define server logic required to draw a histogram
+server <- function(input, output) {
+    
+    ids <- data.frame(forecaster_id = as.character(c("332193", "991083", "424666", 
+                                                  "188218", "492860", "419395")), 
```

Comment:
Assume this will need linking to the user csv when/if opening this to the public.

Reply:
ooh yes - forgot I even included the performance-board in the app. That needs a rework

Reply:
😄